de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -1014,12 +1016,16 @@ public void extendContract(final @NonNull ContractExtendingRequest context)
 	{
 		Services.get(ITrxManager.class).run(ITrx.TRXNAME_ThreadInherited, localTrxName -> {
 
+			final Set<Integer> conditionsBreadCrumb = new HashSet<>();
```

Comment:
why not `seenFlatrateConditionIds` instead of `conditionsBreadCrumb` ?

Reply:
ok. I named it like because in my head I was thinking to a breadcrumb